src/org/rust/lang/core/resolve/RustResolveEngine.kt

```diff
@@ -68,6 +79,26 @@ public class RustResolveEngine() {
             seek(scope.getDeclarations())
         }
 
+        private fun processUseDeclaration(use: RustUseItem) {
+            val path = use.viewPath
+            val pathPart = path.pathPart ?: return
+
+            val isPlainPathImport = path.mul == null && path.lbrace == null
+            if (isPlainPathImport) {
+                val name = path.rename ?: pathPart
+                if (match(name)) {
+                    if (use in visited) {
```

Comment:
It's kinda misleading, that we're checking whether we're already visited node in the middle of the method instead of immediate halt.